datafusion/functions/src/core/map.rs

```diff
@@ -235,6 +203,25 @@ impl ScalarUDFImpl for MakeMap {
             }
         }
 
+        let mut result = Vec::new();
+        for _ in 0..arg_types.len() / 2 {
+            result.push(key_type.clone());
+            result.push(value_type.clone());
```

Comment:
It seems there is no coercion here, in this case we don't even need `coerce_types`.
Shouldn't we convert null to non-null here?

Reply:
No, I think it works well because the following case is passed.
```
query ?
SELECT MAP(['POST', 'HEAD', 'PATCH'], [41, 33, null]);
----
{POST: 41, HEAD: 33, PATCH: }
```

Because the type of all values or all keys should be the same, after the type checking, I build a new vec based on the final key type and final value type. In this case, the original types are
```
[DataType::Utf8, DataType::Int32, DataType::Utf8,, DataType::Int32, DataType::Utf8,, DataType::Null]
```
It would be transformed to 
```
[DataType::Utf8, DataType::Int32, DataType::Utf8,, DataType::Int32, DataType::Utf8,, DataType::Int32]
```

So, I think we can get `Int32(Null)` when invoking, and then create a type-correct null array for it.

Reply:
Ok, I see the code above L206, it was collapsed so I miss it